### PR TITLE
Base Viridian Support [CLC-151]

### PR DESCRIPTION
**Pull request by yuce**
This PR:
* Adds support for API key/secret and email/password authentication to Viridian.
* Adds `viridian login` and `viridian list-clusters` commands.
* Adds the initial support for reading and writing secrets.
* Adds the corresponding documentation

### CODE REVIEW

**Review by kutluhanmetin — DISMISSED** [review]

**Review by mtyazici — APPROVED** [review]
LGTM, I just left a comment. Because this PR is urgent we can maybe improve it in another PR.

File: clc/shell/prompt.go
[the remark this answers is not in the document]
Comment (yuce) [reply clc/shell/prompt.go:1]: Good idea. I've updated the prompter and used it for the API prompt at: https://github.com/hazelcast/hazelcast-commandline-client/pull/222/commits/bfe3549be4df8549220f246ad007d8d82828d84f

**Comment by yuce** [comment]
@mtyazici 
>  Because this PR is urgent we can maybe improve it in another PR.
If there are any problems other than tests it would be better to fix them in this PR.

File: clc/shell/prompt.go
[the remark this answers is not in the document]
Comment (mtyazici) [reply clc/shell/prompt.go:1]: I think we did not do the same for the legacy login prompts. Link [here](https://github.com/hazelcast/hazelcast-commandline-client/pull/222/files#diff-89763d072723a7e2798875d3f5f037c8660575d13f8fc62ec610641ff855f6c7R148). Are we planning to use `clc/shell/prompt.go`?

[the remark this answers is not in the document]
Comment (yuce) [reply clc/shell/prompt.go:1]: I've removed legacy login at: https://github.com/hazelcast/hazelcast-commandline-client/pull/222/commits/9a61a6d368291228472a7bb2c18d362527768a83

**Review by mtyazici — APPROVED** [review]

**Review by kutluhanmetin — APPROVED** [review]

**Pull request merged**